agent/api/grpc/server.go

```diff
@@ -235,3 +241,57 @@ func (s *grpcServer) Attestation(req *agent.AttestationRequest, stream agent.Age
 
 	return nil
 }
+
+func decodeIMAMeasurementsRequest(_ context.Context, grpcReq interface{}) (interface{}, error) {
+	return imaMeasurementsReq{}, nil
+}
+
+func encodeIMAMeasurementsResponse(_ context.Context, response interface{}) (interface{}, error) {
+	res := response.(imaMeasurementsRes)
+	return &agent.IMAMeasurementsResponse{
+		File:  res.File,
+		Pcr10: res.PCR10,
+	}, nil
+}
+
+func (s *grpcServer) IMAMeasurements(req *agent.IMAMeasurementsRequest, stream agent.AgentService_IMAMeasurementsServer) error {
+	_, res, err := s.imaMeasurements.ServeGRPC(stream.Context(), req)
+	if err != nil {
+		return err
+	}
+	rr := res.(*agent.IMAMeasurementsResponse)
+
+	if err := stream.SetHeader(metadata.New(map[string]string{FileSizeKey: fmt.Sprint(len(rr.File))})); err != nil {
+		return status.Error(codes.Internal, err.Error())
+	}
+
+	imaMeasurementsBuffer := bytes.NewBuffer(rr.File)
+	pcr10Buffer := bytes.NewBuffer(rr.Pcr10)
+
+	imaMeasurementsResultBuffer := make([]byte, bufferSize)
+	pcr10ResultBuffer := make([]byte, bufferSize)
+
+	for {
+		iman, err := imaMeasurementsBuffer.Read(imaMeasurementsResultBuffer)
+		if err == io.EOF {
+			break
+		}
+		if err != nil {
+			return status.Error(codes.Internal, err.Error())
+		}
+
+		pcrn, err := pcr10Buffer.Read(pcr10ResultBuffer)
+		if err == io.EOF {
+			break
+		}
+		if err != nil {
+			return status.Error(codes.Internal, err.Error())
+		}
+
+		if err := stream.Send(&agent.IMAMeasurementsResponse{File: imaMeasurementsResultBuffer[:iman], Pcr10: pcr10ResultBuffer[:pcrn]}); err != nil {
+			return status.Error(codes.Internal, err.Error())
+		}
```

Comment:
Since `rr.File` and `rr.Pcr10` are byte slices, this whole block can be replaced by:
```go
if err := stream.Send(&agent.IMAMeasurementsResponse{File: rr.File, Pcr10: rr.Pcr10}); err != nil {
	return status.Error(codes.Internal, err.Error())
}
```

Reply:
I think this is to read and send in chunks, but this should only be the case if they exceed 4MB in size, the default gRPC limit

Reply:
Got it. In that case, this does make sense but requires some adjustments - 1) shorter, more Go-idiomatic names and 2) proper handling of EOF. If either error is EOF - this code will skip both last chunks.

Reply:
We also need some retires to `Send` if there are no any implicit ones done by the stream.

Reply:
we can add abstractions here for sending chunks from agent

Reply:
If this is the only place we use it, there is no need to abstract anything.  I guess it's impractical due to the way we handle gRPC server responses (we're always creating new types).